src/imu/mpu9255/mpu9255.cpp

```diff
@@ -0,0 +1,150 @@
+#include "mpu9255.hpp"
+
+namespace imu {
+
+auto Mpu9255::Init(void) noexcept -> types::DriverStatus {
+  CreateSensorPointer();
+
+  SetInitConfigMPU9255();
+
+  if (InitAllSensors()) {
+    initialized_ = true;
+    return types::DriverStatus::OK;
+  }
+  return types::DriverStatus::HAL_ERROR;
+}
+
+auto Mpu9255::CreateSensorPointer(void) noexcept -> void {
+  if (gyroscope_ == NULL) {
+    gyroscope_ = std::make_unique<imu::Gyroscope>(i2c_handler_);
+  }
+  if (accelerometer_ == NULL) {
+    accelerometer_ = std::make_unique<imu::Accelerometer>(i2c_handler_);
+  }
+  if (magnetometer_ == NULL) {
+    magnetometer_ = std::make_unique<imu::Magnetometer>(i2c_handler_);
+  }
+  if (temperature_ == NULL) {
+    temperature_ = std::make_unique<imu::Temperature>(i2c_handler_);
+  }
+}
+
+auto Mpu9255::SetInitConfigMPU9255(void) noexcept -> void {
+  SetMPU9255Register(INT_PIN_CFG, 0x22);
+  SetMPU9255Register(INT_ENABLE, 0x01);
+}
+
+auto Mpu9255::SetMPU9255Register(const std::uint8_t register_, const std::uint8_t register_value) noexcept -> void {
```

Comment:
why is the method parameter register_ marked as private by a trailing _?

Reply:
Unfortunately, "register" is c++ internal and keyword: https://en.cppreference.com/w/cpp/keyword/register

Reply:
you are right. I forgot